### PR TITLE
Upgrade to the latest version of webpack by making use of Vuecli 3

### PR DESCRIPTION
**Pull request by mjvkeulen**
## Description
Updated the front end to make use of the latest Webpack and Node environment. Allowing all dependencies to be updated, removing any found vulnerabilities.

### Note
One indirect vulnerability remains `tar`, via `node sass` via `node gyp`. However, this should be resolved soonish see:
https://github.com/nodejs/node-gyp/issues/1714

### CODE REVIEW

File: src/django/VLE/migrations/0021_unknown_profile_picture.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.1.7 on 2019-03-06 11:16
+
+from django.db import migrations, models
+
+
+def update_unknown_profile_pic_url(apps, schema_editor):
+    User = apps.get_model('VLE', 'User')
+    for user in User.objects.all():
+        if user.profile_picture == '/static/unknown-profile.png':
+            user.profile_picture = '/unknown-profile.png'
```
Comment (Larspolo) [inline src/django/VLE/migrations/0021_unknown_profile_picture.py]: Maybe we should set this to `None` and add an alt to the places where we use the pf in the frontend

File: src/vue/package.json [hunk 2]
```diff
@@ -1,33 +1,34 @@
 {
-  "name": "eJournal",
+  "name": "ejournal",
```
Comment (Larspolo) [inline src/vue/package.json]: I prefer eJournal

File: src/vue/src/Header.vue [hunk 3]
```diff
@@ -202,7 +202,8 @@ export default {
         &:hover
             > svg
                 fill: $theme-medium-grey !important
-        &.active
```
Comment (Larspolo) [inline src/vue/src/Header.vue]: The active isnt orange anymore

File: src/vue/src/views/UserRoleConfiguration.vue [hunk 4]
```diff
@@ -48,7 +48,8 @@
                             <!-- eslint-disable -->
                             <b-form-checkbox
                                 :class="{ 'input-disabled': essentialPermission(role, permission) }"
-                                v-model="roleConfig[getIndex(role)][permission]"/>
+                                v-model="roleConfig[getIndex(role)][permission]"
+                                inline/>
```
Comment (Larspolo) [inline src/vue/src/views/UserRoleConfiguration.vue]: shouldnt this `/>` be on a newline?

File: src/vue/package.json (hunk 2 above)
Comment (mjvkeulen) [reply src/vue/package.json]: Should be non capitalized

File: src/vue/src/Header.vue (hunk 3 above)
Comment (mjvkeulen) [reply src/vue/src/Header.vue]: This was specifically addressed in 1850ff18f452457e59ef6d3e729e54ed23b7a3d6, are you sure?

File: src/vue/src/views/UserRoleConfiguration.vue (hunk 4 above)
Comment (mjvkeulen) [reply src/vue/src/views/UserRoleConfiguration.vue]: Yes, hm why is this not caught by the linter tho

File: src/vue/src/Header.vue (hunk 3 above)
Comment (mjvkeulen) [reply src/vue/src/Header.vue]: Ah the behaviour is different in the way that currently only the exact match is highlighted. Is this prefered or not? So should courses be highlighted when viewing a journal page, or should this only happen when visiting the view `Course`?

This change is as simple as moving from `router-link-exact-active` to `router-link-active` in the selector.

Comment (engelhamer) [reply src/vue/src/Header.vue]: I'd be in favor of `router-link-active`, just like how it used to be.

Comment (mjvkeulen) [reply src/vue/src/Header.vue]: Alright, I will update accordingly

File: src/vue/src/views/UserRoleConfiguration.vue (hunk 4 above)
Comment (mjvkeulen) [reply src/vue/src/views/UserRoleConfiguration.vue]: Ah silly this block is disabled, for syntax which is no longer disallowed even.

**Comment by mjvkeulen** [comment]
fixes #603

**Pull request merged**